extensions.c

```diff
@@ -263,14 +263,18 @@ hibaext_maybe_deflate(struct sshbuf *in, int compress, struct sshbuf *out) {
 			}
 			sshbuf_put(temp, buf, stream.total_out);
 		}
+		deflateEnd(&stream);
 		ret = HIBA_OK;
 		debug3("hibaext_maybe_deflate: compression out %zu.", sshbuf_len(temp));
 	}
 err:
 #endif  /* WITH_EXTENSION_COMPRESSION */
 
 	if (ret == HIBA_OK) {
-		return sshbuf_putb(out, temp);
+		ret = sshbuf_putb(out, temp);
```

Comment:
nit: sshbuf_putb returns an SSH error code, whereas hibaext_maybe_deflate returns a HIBA error code. The success code happens to be the same (0), but the error codes do not match (e.g. sshbuf_putb can return SSH_ERR_ALLOC_FAIL = -2, which corresponds to HIBA_BAD_VERSION).